common/lib/xmodule/xmodule/tests/test_content.py

```diff
@@ -173,3 +179,101 @@ def test_static_content_write_js(self):
         js_file_paths = [file_path for file_path in js_file_paths if os.path.basename(file_path).startswith('000-')]
         self.assertEqual(len(js_file_paths), 1)
         self.assertIn("XModule.Descriptor = (function () {", open(js_file_paths[0]).read())
+
+
+@ddt.ddt
+class CanonicalContentTest(ModuleStoreTestCase):
+    """
+    Tests the generation of canonical asset URLs for different types
+    of assets: c4x-style, opaque key style, locked, unlocked, CDN
+    set, CDN not set, etc.
+    """
+
+    def setUp(self):
+        super(CanonicalContentTest, self).setUp()
+
+        with self.store.default_store(ModuleStoreEnum.Type.split):
+            self.new_course = CourseFactory.create(org='a', course='b', run='d')
+
+            # Create an unlocked image.
+            unlocked_buf = StringIO()
+            unlocked_name = "n_unl.png"
+
+            # Save the course image to the content store.
+            unlocked_asset_key = StaticContent.compute_location(self.new_course.id, unlocked_name)
+            unlocked_content = StaticContent(unlocked_asset_key, unlocked_name, 'image/png', unlocked_buf)
+            contentstore().save(unlocked_content)
+
+            # Create a locked image.
+            locked_buf = StringIO()
+            locked_name = "n_l.png"
```

Comment:
Please give more descriptive file names. I realize what you mean here, but I suspect that someday someone's going to come across a very random error in a test using a crappy console font, think it means PNG number 1, and scratch their heads. ;-)